### PR TITLE
remove scala bindings and ledger-api-client

### PR DESCRIPTION
**Pull request by mziolekda**

### CODE REVIEW

File: compatibility/maven_install.json
Comment (samuel-williams-da) [inline compatibility/maven_install.json:1]: I'm not familiar with the functionality of this file, so can't give it a thorough review

File: language-support/java/bindings-rxjava/BUILD.bazel [hunk 1]
```diff
@@ -36,7 +36,7 @@ da_java_library(
     ],
     deps = [
         "//canton:bindings-java",
-        "//ledger/ledger-api-auth-client",
+        #"//canton:community_ledger_ledger-api-core",
```
Comment (samuel-williams-da) [inline language-support/java/bindings-rxjava/BUILD.bazel:39]: Is this needed?

File: language-support/java/bindings-rxjava/src/main/java/com/daml/ledger/rxjava/grpc/helpers/LedgerCallCredentials.java [hunk 2]
```diff
@@ -1,13 +1,14 @@
...
 
-package com.daml.ledger.api.auth.client;
+package com.daml.ledger.rxjava.grpc.helpers;
 
 import io.grpc.CallCredentials;
 import io.grpc.Metadata;
 import io.grpc.stub.AbstractStub;
 import java.util.concurrent.Executor;
 
+@SuppressWarnings("deprecation")
```
Comment (samuel-williams-da) [inline language-support/java/bindings-rxjava/src/main/java/com/daml/ledger/rxjava/grpc/helpers/LedgerCallCredentials.java:11]: What here is deprecated? Should there be a TODO around here somewhere?

File: language-support/java/bindings-rxjava/src/test/scala/com/daml/ledger/rxjava/package.scala [hunk 3]
```diff
@@ -3,43 +3,35 @@
...
+//  private[rxjava] val authorizer =
+//    new Authorizer(
+//      () => Clock.systemUTC().instant(),
+//      "testLedgerId",
+//      "testParticipantId",
+//      new InMemoryUserManagementStore(NamedLoggerFactory.root),
+//      ExecutionContext.parasitic,
+//      userRightsCheckIntervalInSeconds = 1,
+//      pekkoScheduler = pekkoSystem.scheduler,
+//    )(LoggingContext.ForTesting)
```
Comment (samuel-williams-da) [inline language-support/java/bindings-rxjava/src/test/scala/com/daml/ledger/rxjava/package.scala:34]: I see you moved from this authorizer to one in `ledgerServices`
Are they the same? Can this be properly removed?

**Review by samuel-williams-da — APPROVED** [review]
Wow! Farewell to all that code :wave: 
I can't speak for the maven file, but the rest looks good :)

File: language-support/java/bindings-rxjava/BUILD.bazel (hunk 1 above)
Comment (mziolekda) [reply language-support/java/bindings-rxjava/BUILD.bazel:39]: well spotted

File: compatibility/maven_install.json
Comment (mziolekda) [reply compatibility/maven_install.json:1]: it is __AUTOGENERATED_FILE_DO_NOT_MODIFY_THIS_FILE_MANUALLY

Comment (samuel-williams-da) [reply compatibility/maven_install.json:1]: Ah right, I assume its fine then.

File: language-support/java/bindings-rxjava/src/main/java/com/daml/ledger/rxjava/grpc/helpers/LedgerCallCredentials.java (hunk 2 above)
Comment (mziolekda) [reply language-support/java/bindings-rxjava/src/main/java/com/daml/ledger/rxjava/grpc/helpers/LedgerCallCredentials.java:11]: It was some fancy instability signalling by the grpc dev team: https://grpc.github.io/grpc-java/javadoc/io/grpc/CallCredentials.html#thisUsesUnstableApi()

Seems ok to remove it now

**Pull request merged**